src/app/lists/ListDataview.tsx

```diff
@@ -0,0 +1,195 @@
+"use client";
+
+import Table, { Datum, RowData } from "@/components/Tables/Table";
+import React, { useState } from "react";
+import styled from "styled-components";
+import EditModal, { EditModalState } from "@/app/lists/EditModal";
+import supabase, { Schema } from "@/supabase";
+import ConfirmDialog from "@/components/Modal/Confirm";
+import Snackbar from "@mui/material/Snackbar/Snackbar";
+import Alert from "@mui/material/Alert/Alert";
+
+type ListRow = Schema["lists"];
+
+interface Props {
+    data: ListRow[];
+}
+
+export const headers: [string, string][] = [
+    ["item_name", "Description"],
+    ["1_quantity", "Single"],
+    ["2_quantity", "Family of 2"],
+    ["3_quantity", "Family of 3"],
+    ["4_quantity", "Family of 4"],
+    ["5_quantity", "Family of 5"],
+    ["6_quantity", "Family of 6"],
+    ["7_quantity", "Family of 7"],
+    ["8_quantity", "Family of 8"],
+    ["9_quantity", "Family of 9"],
+    ["10_quantity", "Family of 10+"],
+];
+
+export const tooltips: [string, string][] = [
+    ["1_notes", "Single"],
+    ["2_notes", "Family of 2"],
+    ["3_notes", "Family of 3"],
+    ["4_notes", "Family of 4"],
+    ["5_notes", "Family of 5"],
+    ["6_notes", "Family of 6"],
+    ["7_notes", "Family of 7"],
+    ["8_notes", "Family of 8"],
+    ["9_notes", "Family of 9"],
+    ["10_notes", "Family of 10+"],
+];
+
+const ListsDataView: React.FC<Props> = ({ data: rawData }) => {
+    const [modal, setModal] = useState<EditModalState>();
+    const [toDelete, setToDelete] = useState<number | null>(null);
+    // need another setState otherwise the modal content changes before the close animation finishes
+    const [toDeleteModalOpen, setToDeleteModalOpen] = useState<boolean>(false);
+    const [errorMsg, setErrorMsg] = useState<string | null>(null);
+
+    if (rawData === null) {
+        throw new Error("No data found");
```

Comment:
How is this error handled? Is it catched?

Reply:
It'll show the nextjs error page